src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticEnabledTest.java

```diff
@@ -309,6 +305,59 @@ void saveFormDataUniqueDocumentIdTest() throws IOException
     }
 
+    @Test
+    void updateFormDataTest() throws JsonProcessingException
+    {
+        doReturn(userList).when(mockUserDetails).getUserDetails();
+        Map<String, Object> testFormData = new HashMap<>();
+        testFormData.put(NAME, NAME_VALUE);
+        testFormData.put(AGE,AGE_VALUE);
+        testFormData.put(ID,EMPTY_STRING);
+        Map<String, Object> testFormMetaData = new HashMap<>();
+        testFormMetaData.put(FORM_VERSION, 1);
+        FormDataSchema formDataSchemaTest=new FormDataSchema(TEST_ID_VALUE,TEST_FORM_ID,
+                TEST_VERSION,testFormData,testFormMetaData);
+        Mockito.when(mockMongoTemplate.collectionExists(anyString())).thenReturn(true);
+        LinkedHashMap<String, Object> formDataMap = new LinkedHashMap<>();
+        formDataMap.put(UNDERSCORE_ID,Long.parseLong(UNDERSCORE_ID_VALUE));
+        formDataMap.put(FORM_ID,TEST_FORM_ID);
+        formDataMap.put(VERSION, String.valueOf(1));
+        formDataMap.put(FORM_META_DATA, formDataSchemaTest.getFormMetaData());
+        formDataMap.put(FORM_DATA, formDataSchemaTest.getFormData());
+        formDataMap.put(CREATED_ON, Date.from(Instant.now()));
+        formDataMap.put(CREATED_BY_ID,CREATED_BY_USER_ID);
+        formDataMap.put(CREATED_BY_NAME, CREATED_BY_USER_NAME);
+        formDataMap.put(UPDATED_ON,TEST_UPDATED_ON);
+        formDataMap.put(UPDATED_BY_ID,UPDATED_BY_USER_ID);
+        formDataMap.put(UPDATED_BY_NAME,UPDATED_BY_USER_NAME);
+        Document document = new Document();
+        document.put(UNDERSCORE_ID,TEST_ID_VALUE);
+        document.put(FORM_DATA,testFormData);
+        document.put(FORM_META_DATA,testFormMetaData);
+        document.put(VERSION,1);
+        FindIterable<Document> iterable = mock(FindIterable.class);
+        MongoCursor cursor = mock(MongoCursor.class);
+        Mockito.when(mockMongoTemplate.getCollection(anyString())).thenReturn(mockMongoCollection);
+        Mockito.when(mockMongoCollection.find((Bson) any())).thenReturn(iterable);
+        Mockito.when(iterable.iterator()).thenReturn(cursor);
+        Mockito.when(cursor.hasNext()).thenReturn(true);
+        Mockito.when(cursor.next()).thenReturn(document);
+
+
+        ReflectionTestUtils.setField(mockFormDataServiceImpl, ELASTIC_ENABLE, true);
+        Mockito.when(mockTokenUtils.getTokenFromContext()).thenReturn(TEST_TOKEN);
+        String postResponse="{\n" +
+                "    \"data\": {\n" +
+                "        \"id\": \"963403130239434752\",\n" +
+                "        \"version\": 1\n" +
+                "    },\n" +
+                "    \"success\": true,\n" +
+                "    \"message\": \"Form data saved successfully\"\n" +
+                "}";
+        Mockito.when(mockWebClientWrapper.webclientRequest(any(),any(),eq(POST),any())).thenReturn(postResponse);
+        Assertions.assertNotNull(mockFormDataServiceImpl.updateFormData(formDataSchemaTest));
```

Comment:
why is update form data test case impacted?

Reply:
No it was not impacted i have covered it to reach target of 85% code coverage which was not previously covered